osgp-protocol-adapter-dlms/src/main/java/org/osgp/adapter/protocol/dlms/application/services/DomainHelperService.java

```
@@ -67,75 +67,57 @@ public DlmsDevice findDlmsDevice(final String deviceIdentification) throws Funct
-
-        // If the result is null then the meter is not in session (not
-        // awake).
-        // So wake up the meter and start polling for the session
-        if (deviceIpAddress == null) {
-            this.jasperWirelessSmsClient.sendWakeUpSMS(iccId);
+            // If the result is null then the meter is not in session (not
+            // awake).
+            // So wake up the meter and start polling for the session
+            this.jasperWirelessSmsClient.sendWakeUpSMS(dlmsDevice.getIccId());
             for (int i = 0; i < this.jasperGetSessionRetries; i++) {
-                try {
-                    Thread.sleep(this.jasperGetSessionSleepBetweenRetries);
-                } catch (final InterruptedException e) {
-                    throw new SessionProviderException(
-                            "Interrupted while sleeping before calling the sessionProvider.getIpAddress", e);
-                }
-                deviceIpAddress = sessionProvider.getIpAddress(iccId);
+                Thread.sleep(this.jasperGetSessionSleepBetweenRetries);
+                deviceIpAddress = sessionProvider.getIpAddress(dlmsDevice.getIccId());
+
                 if (deviceIpAddress != null) {
                     return deviceIpAddress;
                 }
             }
+        } catch (final InterruptedException e) {
+            throw new ProtocolAdapterException(
+                    "Interrupted while sleeping before calling the sessionProvider.getIpAddress", e);
+        } catch (SessionProviderUnsupportedException | SessionProviderException e) {
+            throw new ProtocolAdapterException("", e);
+        }
+        if (deviceIpAddress == null || "".equals(deviceIpAddress)) {
+            LOGGER.error("Session provider: {} did not return an IP address for device: {} and iccId: {}",
+                    dlmsDevice.getCommunicationProvider(), dlmsDevice.getDeviceIdentification(), dlmsDevice.getIccId());
```

I think it is better to throw an exception early on. At this point we know the connection is not going to be set up because determining the IP address failed. In the code before the changes related to the static IP address an exception indeed was thrown.